projects/electron-ngrx/src/lib/window-communication.service.ts

```diff
@@ -71,6 +76,80 @@ export class WindowCommunicationService {
     return this.getReplayByMessageId<R>(messageId);
   }
 
+  listenSubscriptionRequest<T>() {
+    return this.subscription.pipe(map((message: { data: T, messageId: number, senderId: number }) => ({
+      data: message.data,
+      response: (observable: Observable<any>) => {
+        const channelPrefix = channel.subscription + message.messageId;
+        const cleanup = new Subject();
```

Comment:
why is this a subject - it is an observable, instead of using in the subscribe make it equal to the onWindowClose